### PR TITLE
Feat sparo branch

### PR DESCRIPTION
**Pull request by chengcyber**
### Basic Checks

Have you run `rush change` for this change?

- [x] Yes
- [ ] No

If **No**, please run `rush change` before, this is necessary.

If adding a **new feature**, the PR's description includes:

- [ ] Reason for adding this feature
- [ ] How to use
- [ ] A basic example

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

### Summary

Add a new enhanced command "sparo branch".

### Detail

When running "sparo branch -d/-D <branch>", it will automatically clean up the remote fetch git configuation. To prevent a error when running "pull" command when the tracking branch is deleted remotely(such as merged into the main branch).

### How to test it

Local

### CODE REVIEW

File: apps/website/docs/pages/commands/overview.md [hunk 1]
```diff
@@ -13,12 +13,14 @@ Sparo has four kinds of subcommands:
    - `sparo clone`
    - `sparo fetch`
    - `sparo pull`
```
Comment (octogonz) [inline apps/website/docs/pages/commands/overview.md:15]: Looks like the website does not reflect the current `overview.md` file:
image

We need to remember to manually redeploy after updating the website, or else set up a GitHub Action for automated deployments.

File: apps/website/docs/pages/commands/overview.md [hunk 2]
```diff
@@ -13,12 +13,14 @@ Sparo has four kinds of subcommands:
...
+   - `sparo branch`
 
 3. **Renamed subcommands** are the mirrored versions of the four enhanced subcommands. They are renamed to add a `git-` prefix:
   - `sparo git-checkout`
   - `sparo git-clone`
   - `sparo git-fetch`
   - `sparo git-pull`
+  - `sparo git-branch`
```
Comment (octogonz) [inline apps/website/docs/pages/commands/overview.md:23]: The list above is supposed to be alphabetized.

File: apps/website/docs/pages/commands/overview.md [hunk 3]
```diff
@@ -13,12 +13,14 @@ Sparo has four kinds of subcommands:
    - `sparo clone`
```
Comment (octogonz) [inline apps/website/docs/pages/commands/overview.md:13]: The text above says `There are four enhanced commands:` -- seems like now there are 5

File: common/changes/sparo/feat-sparo-branch_2024-05-21-23-13.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "sparo",
+      "comment": "\"sparo branch -d\" clean up remote fetch git configuration",
```
Comment (octogonz) [inline common/changes/sparo/feat-sparo-branch_2024-05-21-23-13.json:5]:
```suggestion
      "comment": "\"sparo branch -d\" clean up remote fetch git configuration",
```
This sentence is not understandable to someone unfamiliar with your PR.  [Review these docs](https://rushjs.io/pages/best_practices/change_logs/) for tips about authoring change messages.

File: common/changes/sparo/feat-sparo-branch_2024-05-21-23-18.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "sparo",
+      "comment": "Add a renamed command \"sparo git-branch\"",
```
Comment (octogonz) [inline common/changes/sparo/feat-sparo-branch_2024-05-21-23-18.json:5]:
```suggestion
      "comment": "Add a mirrored command \"sparo git-branch\"",
```

The docs call them [mirrored commands](https://tiktok.github.io/sparo/pages/commands/overview/)

**Comment by chengcyber** [comment]
I had a session with Pete. We conclude that we are leaning to the approach to clean up git configuration during `sparo fetch` and `sparo pull`.

Closing this PR in flavor of https://github.com/tiktok/sparo/pull/72

**Pull request closed without merging**